cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -620,7 +629,7 @@ where
 			} else if local_trace_idx == local_receipt.trace.len() - 1 {
 				// `finalize_block` execution proof.
 				let pre_state_root = as_h256(&execution_receipt.trace[local_trace_idx - 1])?;
-				let post_state_root = as_h256(&execution_receipt.trace[local_trace_idx])?;
+				let post_state_root = as_h256(local_root)?;
```

Comment:
This is good, but there is still a bunch of indexing left, would be nice to not have it or have `.expect` that explains why it can never ever happen.

Reply:
There are two index expressions left, both of them are `trace[local_trace_idx - 1]`, they will never panic since the index was from the output of `enumerate()`(means it will never overflow) and `local_trace_idx == 0` has been handled particularly.  I thought they are already enough explicit and using the index expression is much simpler, do you still want a `expect` here to explain this?

Reply:
Yeah, I also admire the explicitness, but given that it logically can't be wrong, I also prefer the simplicity :D.